packages/airnode-deployer/src/infrastructure/aws.ts

```diff
@@ -1,74 +1,203 @@
+import * as fs from 'fs';
 import AWS from 'aws-sdk';
+import concat from 'lodash/concat';
+import compact from 'lodash/compact';
 import { AwsCloudProvider } from '@api3/airnode-node';
 import { go } from '@api3/promise-utils';
 import * as logger from '../utils/logger';
+import {
+  BUCKET_NAME_REGEX,
+  Directory,
+  FileSystemType,
+  generateBucketName,
+  translatePathsToDirectoryStructure,
+} from '../utils/infrastructure';
 
-async function deleteObjects(s3: AWS.S3, bucket: string, objects: AWS.S3.ObjectIdentifierList) {
-  logger.debug(`Deleting objects ${JSON.stringify(objects)} from S3 bucket ${bucket}`);
-  await s3
-    .deleteObjects({
-      Bucket: bucket,
-      Delete: {
-        Objects: objects.map(({ Key, VersionId }) => ({ Key, VersionId })),
-      },
-    })
-    .promise();
-}
-
-async function fetchObjects(s3: AWS.S3, bucket: string) {
-  logger.debug(`Fetching objects from S3 bucket ${bucket}`);
-  const objectResults = await s3.listObjectVersions({ Bucket: bucket }).promise();
-  const objects = [...(objectResults.Versions ?? []), ...(objectResults.DeleteMarkers ?? [])];
-  return objects.filter((object): object is AWS.S3.ObjectIdentifier => !!object.Key);
-}
-
-async function emptyBucket(s3: AWS.S3, bucket: string) {
-  logger.debug(`Emptying S3 bucket ${bucket}`);
-  let objects = await fetchObjects(s3, bucket);
-
-  // At most 1000 objects are fetched in one network request
-  while (objects.length > 0) {
-    await deleteObjects(s3, bucket, objects);
-
-    objects = await fetchObjects(s3, bucket);
+const initializeS3Service = (cloudProvider: AwsCloudProvider) => {
+  AWS.config.update({ region: cloudProvider.region });
+  return new AWS.S3();
+};
+
+export const getAirnodeBucket = async (cloudProvider: AwsCloudProvider) => {
+  const s3 = initializeS3Service(cloudProvider);
+
+  logger.debug('Listing S3 buckets');
+  const goBuckets = await go(() => s3.listBuckets().promise());
+  if (!goBuckets.success) {
+    throw new Error(`Failed to list S3 buckets: ${goBuckets.error}`);
+  }
+
+  const airnodeBuckets = goBuckets.data.Buckets?.filter((bucket) => bucket.Name?.match(BUCKET_NAME_REGEX));
```

Comment:
I'd add an 
```
if (!airnodeBuckets) throw new Error('...')
```

to avoid checking for undefined below.

Reply:
I'm pretty sure that's just bad typing on the AWS side. They type everything as optional even though it has to be in the response.